deep_qa/run.py

```diff
@@ -0,0 +1,160 @@
+from typing import Union
+import sys
+import logging
+import shutil
+
+import random
+import pyhocon
+import numpy
+
+# pylint: disable=wrong-import-position
+from deep_qa.common.params import Params, replace_none, ConfigurationError
+from deep_qa.common.tee_logger import TeeLogger
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+def prepare_environment(params: Union[Params, dict]):
+    """
+    Sets random seeds for reproducible experiments. This may not work as expected
+    if you use this from within a python project in which you have already imported Keras.
+    If you use the scripts/run_model.py entry point to training models with this library,
+    your experiments should be reproducible. If you are using this from your own project,
+    you will want to call this function before importing Keras.
+
+     Parameters
+    ----------
+    params: ``Params`` object or dict, required.
+        A ``Params`` object or dict holding the json parameters.
+    """
+    seed = params.pop("random_seed", 13370)
+    numpy_seed = params.pop("numpy_seed", 1337)
+    if "keras" in sys.modules:
+        logger.warning("You have already imported Keras in your code. If you are using DeepQA"
+                       "functionality to set random seeds, they will have no effect, as code"
+                       "prior to calling this function will be non-deterministic. We will not"
+                       "the random seed here.")
+        seed = None
+        numpy_seed = None
+    if seed is not None:
+        random.seed(seed)
+    if numpy_seed is not None:
+        numpy.random.seed(numpy_seed)
+
+    from deep_qa.common.checks import log_keras_version_info
+    log_keras_version_info()
+
+
+def run_model(param_path: str, model_class=None):
+    """
+    This function is the normal entry point to DeepQA. Use this to run a DeepQA model in
```

Comment:
How about this for the first two sentences: "This function is the normal entry point to DeepQA.  It typically trains, validates, and tests the model specified in the parameters in `param_path`, but can do a subset of that, depending on the files included in the parameters."

And now that I look more at `model.train()`, the more I think that calling this `train_model` actually does make sense...  Sorry, I'm going back and forth here.  I say we should keep it like this for now, but maybe refactor things in the future, so that the testing functionality is moved out of `model.train()`, and then this can be called `train_model`.